sam/singleCell.go

```diff
@@ -0,0 +1,50 @@
+package sam
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+	"strings"
+)
+
+//SingleCellAlignment includes a Sam struct along with the parsed barcode and UMI for single-cell reads.
+type SingleCellAlignment struct {
+	Aln Sam
+	Bx  []dna.Base
+	Umi []dna.Base
+}
+
+//ToSingleCellAlignment parses the barcode and UMI from the QNAME field of an input sam from a single-cell read formatted with fastqFormat -singleCell.
+func ToSingleCellAlignment(s Sam) SingleCellAlignment {
```

Comment:
I wonder what is the benefit of having ToSingleCellAlignment when it looks like it could have just been a small addition to parseBxAndUmiFromAln?